### ISSUE TITLE
[Enhancement]: Add X and Y bounds to 2d coordinates

### ISSUE THREAD

**Issue by acordonez** [xCDAT/xcdat#676]
### Is your feature request related to a problem?

I don't see any similar issues. I would like the bounds.add_missing_bounds() function to be able to add bounds to two-dimensional coordinates.

### Describe the solution you'd like

I've added a code example here to demonstrate.

```
import cf_xarray
import numpy as np
import xarray as xr
import xcdat as xc

# Create 2-d lat and lon arrays
lat=np.arange(90,-91,-5)
lon=np.arange(0,361,5)
lon = np.transpose(np.broadcast_to(np.squeeze(lon),(37,73)))
lat = np.transpose(np.transpose(np.broadcast_to(lat,(73,37))))
data = np.random.rand(73,37)

ds = xr.DataArray(data=data, coords={"lon":(["x","y"],lon),"lat":(["x","y"],lat)}, dims=["x","y"], attrs=None)
ds = ds.to_dataset(name="data")

# The printed result will not have lat lon bounds
ds.bounds.add_missing_bounds()

# The printed result will include lat lon bounds
# I would like to get this result using xcdat.bounds.add_missing_bounds()
ds.cf.add_bounds(["lat","lon"])
```
Current result with xcdat:
![Screenshot 2024-07-02 at 3 11 10 PM](https://github.com/xCDAT/xcdat/assets/18147700/ccd36bf3-1031-4445-abad-9296130adfac)
Desired result:
![Screenshot 2024-07-02 at 3 11 24 PM](https://github.com/xCDAT/xcdat/assets/18147700/5b2005b3-49a2-4735-9d77-f216d89ce81c)

### Describe alternatives you've considered

This can be done with cf_xarray, as I've shown in my example. Because so many xcdat features require datasets to have bounds, however, I think it would be very helpful for xcdat to be able to add those bounds without asking the user to research and incorporate another library.

Another alternative would be to point users to cf_xarray for 2d bounds in the add_missing_bounds() documentation.

### Additional context

Two dimensional coordinates are very common in ocean and sea ice datasets.

**Comment by pochedls** [comment]
@acordonez – could you point to an example dataset where this exists (or a dataset where this behavior would apply if you opened the dataset and then dropped the bounds via `ds = ds.drop_vars('lat_bnds')`?

A couple things that might be considered for this enhancement:
* Should xcdat just call the cf-xarray functionality to add bounds (for this feature and in general)? 
* What happens now if someone tried to regrid this dataset with xcdat + xesmf? I'm curious if the bounds are not needed, are generated, or if xcdat can't currently regrid this dataset (unless the user adds the bounds with cf-xarray). Aside from regridding, other spatial functionality that uses bounds (e.g., spatial averaging), does not work on non-rectilinear data.

**Comment by acordonez** [comment]
@pochedls Regarding regridding, in my experience xcdat + xesmf cannot regrid this dataset without the bounds. I actually stumbled across this by trying to regrid a non-rectilinear dataset. 
The dataset I was working with is on NERSC, but I can try to find something local to point you to.

**Comment by pochedls** [comment]
> @pochedls Regarding regridding, in my experience xcdat + xesmf cannot regrid this dataset without the bounds. I actually stumbled across this by trying to regrid a non-rectilinear dataset. The dataset I was working with is on NERSC, but I can try to find something local to point you to.

Would a dataset like this work (if bounds are dropped): `/p/css03/esgf_publish/CMIP6/CMIP/CSIRO/ACCESS-ESM1-5/historical/r10i1p1f1/SImon/siconc/gn/v20200817/siconc_SImon_ACCESS-ESM1-5_historical_r10i1p1f1_gn_185001-201412.nc`

Thank you for clarifying on regridding. Since this issue would prevent a user from regridding, it seems like it would be useful to add functionality to add bounds for 2D coordinates.

**Comment by acordonez** [comment]
> /p/css03/esgf_publish/CMIP6/CMIP/CSIRO/ACCESS-ESM1-5/historical/r10i1p1f1/SImon/siconc/gn/v20200817/siconc_SImon_ACCESS-ESM1-5_historical_r10i1p1f1_gn_185001-201412.nc

@pochedls Yes it looks like the file you suggested would reproduce this behavior if the bounds were dropped.
